presto-main/src/test/java/com/facebook/presto/execution/resourceGroups/TestResourceGroups.java

```diff
@@ -390,6 +385,153 @@ public void testWeightedScheduling()
+        assertBetweenInclusive(group2Ran, 1995, 2000);
+    }
+
+    @Test(timeOut = 10_000)
+    public void testWeightedSharesSchedulingEqualWeights()
+    {
+        RootInternalResourceGroup root = new RootInternalResourceGroup("root", (group, export) -> {}, directExecutor());
+        root.setSoftMemoryLimit(new DataSize(1, MEGABYTE));
+        root.setMaxQueuedQueries(50);
+        // Start with zero capacity, so that nothing starts running until we've added all the queries
+        root.setHardConcurrencyLimit(0);
+        root.setSchedulingPolicy(WEIGHTED_SHARES);
+
+        InternalResourceGroup group1 = root.getOrCreateSubGroup("1");
+        group1.setSoftMemoryLimit(new DataSize(1, MEGABYTE));
+        group1.setMaxQueuedQueries(50);
+        group1.setHardConcurrencyLimit(2);
+        group1.setSoftConcurrencyLimit(2);
+        group1.setSchedulingWeight(1);
+
+        InternalResourceGroup group2 = root.getOrCreateSubGroup("2");
+        group2.setSoftMemoryLimit(new DataSize(1, MEGABYTE));
+        group2.setMaxQueuedQueries(50);
+        group2.setHardConcurrencyLimit(2);
+        group2.setSoftConcurrencyLimit(2);
+        group2.setSchedulingWeight(1);
+
+        InternalResourceGroup group3 = root.getOrCreateSubGroup("3");
+        group3.setSoftMemoryLimit(new DataSize(1, MEGABYTE));
+        group3.setMaxQueuedQueries(50);
+        group3.setHardConcurrencyLimit(2);
+        group3.setSoftConcurrencyLimit(2);
+        group3.setSchedulingWeight(2);
+
+        Set<MockQueryExecution> group1Queries = fillGroupTo(group1, ImmutableSet.of(), 4);
+        Set<MockQueryExecution> group2Queries = fillGroupTo(group2, ImmutableSet.of(), 4);
+        Set<MockQueryExecution> group3Queries = fillGroupTo(group3, ImmutableSet.of(), 4);
+        root.setHardConcurrencyLimit(4);
+
+        int group1Ran = 0;
+        int group2Ran = 0;
+        int group3Ran = 0;
+        for (int i = 0; i < 1000; i++) {
+            group1Ran += completeGroupQueries(group1Queries);
+            group2Ran += completeGroupQueries(group2Queries);
+            group3Ran += completeGroupQueries(group3Queries);
+            root.processQueuedQueries();
+            group1Queries = fillGroupTo(group1, group1Queries, 4);
+            group2Queries = fillGroupTo(group2, group2Queries, 4);
+            group3Queries = fillGroupTo(group3, group3Queries, 4);
+        }
+
+        // group 3 should run approximately 2x the number of queries of 1 and 2
+        BinomialDistribution binomial = new BinomialDistribution(4000, 1.0 / 4.0);
+        int lowerBound = binomial.inverseCumulativeProbability(0.000001);
+        int upperBound = binomial.inverseCumulativeProbability(0.999999);
+
+        assertBetweenInclusive(group1Ran, lowerBound, upperBound);
+        assertBetweenInclusive(group2Ran, lowerBound, upperBound);
+        assertBetweenInclusive(group3Ran, 2 * lowerBound, 2 * upperBound);
+    }
+
+    @Test(timeOut = 10_000_000) //TODO: fix
```

Comment:
what's this TODO for? is it related to extremely high timeout?

Reply:
Unintended, reverted.